klog.go

```diff
@@ -873,6 +873,9 @@ func (l *loggingT) output(s severity.Severity, logger *logWriter, buf *buffer.Bu
 		if logger.writeKlogBuffer != nil {
 			logger.writeKlogBuffer(data)
 		} else {
+			if data[len(data)-1] == '\n' {
```

Comment:
I know, `data` can probably never be empty, but perhaps check for it anyway?

Reply:
Yes, it is necessary that

done